airbnb/react-sketchapp
# Handle Sketch Resize Constraints (v44+)

# Overview

This PR adds in full support for sketch v44+ resize constraint controls (without breaking v43 support).
<img width="214" alt="screen shot 2017-09-05 at 1 26 44 am" src="https://user-images.githubusercontent.com/3893845/30047934-57a2981e-91d9-11e7-9e74-177c1ad9ca56.png">

# Changes

- Added new `resizingConstraint` prop to `View`, `Text`, `Image`, and `SymbolInstance` components
  - These props get handled by the component's respective renderers along with `SketchRenderer` base class
- Added new` ResizingConstraintPropTypes` PropType
- Added new` ResizeConstraints ` Flow type
- Updated jest snapshot tests to handle new `resizingConstraint`

# Testing this PR

Since `sketchapp-json-plugin`'s current version wont work, you'll need to setup the plugin locally. Here's how to do it:

1. git clone this repository (`react-sketchapp`), and checkout the `feature/resize_constraints` branch.
2. Install all npm dependencies: `npm install` within the react-sketchapp's root directory and then run `npm run watch`..
3. Navigate to the `react-sketchapp/examples/basic-setup`, open the `package.json` file and swap out the `"react-sketchapp"` with your local clone. It should look something like this: `"react-sketchapp": "file:{path-to-folder}/react-sketchapp",`. Then install all npm packages: `npm install`.
4. Open the `src/my-command.js` file in a text editor and add `resizingConstraint={{bottom: true, fixedWidth: true}}` property to one of the `<Text />` components
5. Open up Sketch (v44+) to a new document and run `npm run render` in the `react-sketchapp` folder within your terminal.
6. Click on the Text object within Sketch (that you added the `resizingConstraint` prop to) and notice that the resizing options panel looks like so:
<img width="214" alt="screen shot 2017-09-05 at 1 26 44 am" src="https://user-images.githubusercontent.com/3893845/30047934-57a2981e-91d9-11e7-9e74-177c1ad9ca56.png">

Congrats 🎉 , you can now edit resize controls via react-sketchapp.

# Related Issues

Closes #150.

# Additional Notes

- This is a first pass API wise, by just allowing a `resizingConstraint={{ top: true, right: true, bottom: true, left: false, fixedHeight: true, fixedWidth: false }}` property on `View`, `Text`, and `Image` components. Definitely open to whatever API makes the most sense here (Props, HOC, etc)
- Prettier decided to format a couple extra files, let me know if that's an issue.
